src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -234,6 +236,7 @@
   <Target Name="CreateVersionFileDuringBuild" Condition="'$(SkipVersionGeneration)'!='true' AND '$(ShouldCreateVersionFileDuringBuild)'=='true'" DependsOnTargets="CreateOrUpdateCurrentVersionFile">
     <PropertyGroup>
       <BuildNumberMajor>$(GeneratedBuildNumberMajor)</BuildNumberMajor>
```

Comment:
Ok makes sense, with my last commit c5e7804 I added the condition to the property group as well.